### ISSUE TITLE
Update ABKFeedController with generics & proper nullability

### ISSUE THREAD

**Issue by jgrandelli** [Appboy/appboy-ios-sdk#78]
Looking at the header for `ABKFeedController` the function `public func getCardsInCategories(categories: ABKCardCategory) -> [AnyObject]?` returns an optional array of any objects. Could this instead be an optional array of `ABKCard`'s. Also the code comment reads:

```
     * @discussion This method will find the cards of given categories and return them.
     * When the given categories don't exist in any card, this method will return an empty array.
```

It would seem then the the function returns an optional array that could _also_ be empty. Is that the case? Could it instead return an optional array that always has contents. Or a non-optional array that is sometimes empty? Because its optional & potentially empty, consumers need to unwrap and check for contents.

**Comment by Wenzhi** [comment]
Hey @jgrandelli,

Sorry for the late response! Your suggestion sounds good to me, I'll talk to the team about it and will let you know as soon as we decide how we are going to change it. 

Thanks,
Wenzhi

**Comment by jgrandelli** [comment]
No worries. Its not terribly pressing.

**Comment by Wenzhi** [comment]
Hey @jgrandelli, 

The new released SDK 2.24.3 has the update you suggested. Please try it out! I'm going to close this issue, but please feel free to re-open it if you still see the issue in the new release.

Thanks,
Wenzhi
